### DISCUSSION TITLE
Showcasing the use of MapLibre in real-world applications

### DISCUSSION THREAD

**Discussion by bchapuis** [maplibre/maplibre#242] [Show and tell]
What are your thoughts on creating a showcase page for the MapLibre website? A really nice example of such page is visible on the [Three.js website](https://threejs.org/).

With the community expanding rapidly, numerous individuals and organizations are building stunning applications with MapLibre. They could benefit from Maplibre's visibility and, conversely, Maplibre may gain a better understanding of how it is utilized in real-world applications. The [show and tell](https://github.com/maplibre/maplibre/discussions/categories/show-and-tell) discussion category has not yet been utilized, but it could serve as a foundation for developing the first version of this page.

![Screenshot 2023-03-22 at 18 17 38](https://user-images.githubusercontent.com/238747/227039397-c5207bab-8b40-441a-955c-bbf1a43e6f08.png)

**Comment by HarelM** [comment]
Currently there's the examples docs and the awesome maplibre repo.
If you would like to make the awesome maplibre repo more interactive I think everyone would benefit from it 😀

**Comment by puppyum** [comment]
**MGuide** ([mguide.app](https://mguide.app)) — Interactive campus map for the University of Michigan.

Built with MapLibre GL JS, it covers 354 buildings with:
- Real-time bus tracking via GTFS-RT protobuf decoding in the browser (both campus and city transit)
- Turn-by-turn walking directions via OSRM
- 8 map themes including CARTO dark-matter, positron, and voyager basemaps
- 3D building extrusions with fill-extrusion layers
- Custom WebGL marker layers for transit vehicles
- Wheelchair-accessible entrance routing
- Fog-of-war exploration game mode using MapLibre paint properties

Uses `map.setStyle()` for theme switching with debounced `styledata` listeners to re-add custom sources/layers after style rebuilds.

**Reply by louwers** [reply]
This is really cool! I love the smooth transition between 2D and 3D.

We don't have a showcase on the website yet but with your permission we can add it to the newsletter sometime.

**Reply by puppyum** [reply]
Thank you for the quick reply! Yeah absolutely, feel free to include it in the newsletter. Happy to provide screenshots or anything else you'd need for that.

**Reply by CommanderStorm** [reply]
Nice work!
Will need to steal some of the features (especially the FOW mode, was unable to test it due to not being in Michigan) for the [TUMs roomfinder](https://nav.tum.de/view/mw).

I like the 3d effect, though does that actually help students navigate.
Curious: What is the idea there?

**Comment by puppyum** [comment]
Haha thanks! It’s supposed to help with entrances on certain floors, if you
select the entrance view you can better understand where it may be,
alongside just looking pretty cool… most people I’ve spoken to about it
seem to enjoy the effect on selection to highlight what building theyre
looking at more prominently!
